test/monitoringlocationlinter/prometheushandler/handler.go

```diff
@@ -0,0 +1,59 @@
+package prometheushandler
+
+import "go/ast"
+
+const (
+	prometheusImportPath = `"github.com/prometheus/client_golang/prometheus"`
+)
+
+// Handler provides different handling, depend on the way prometheus was imported, whether
+// is imported with "." name, custom name or without any name.
+type Handler interface {
+	// IsRegisterMetricsCall checks if a call is a register metrics call
+	IsRegisterMetricsCall(*ast.CallExpr) bool
+}
+
+// GetPrometheusHandler returns a prometheus handler according to the way prometheus was imported in the file.
+func GetPrometheusHandler(file *ast.File) Handler {
```

Comment:
Not sure dot import is a real scenario for prometheus, so I think you only need the name handler. I actually think you don't need the handler at all. The only thing you need from it is the supprot the named import, to make sure the call is from this package.

Reply:
+1, removed the handler.